ocean-client/src/vault-maxi.ts

```diff
@@ -76,17 +76,34 @@ export async function main(event: maxiEvent): Promise<Object> {
     }
 
     let vault: LoanVaultActive = vaultcheck
+
+    // 2022-03-08 Krysh: Something went wrong on last execution, we need to clean up, whatever was done
+    if (settings.state !== VaultMaxiState.Start && settings.state !== VaultMaxiState.Finish) {
+        console.log("something went wrong on last execution, we clean up whatever was done")
+        let result = await program.cleanUp(vault, telegram)
+        await store.updateToState(VaultMaxiState.Start)
+        const cleanUpVaultCheck = await program.getVault() as LoanVaultActive
+        await telegram.log("executed clean-up part of script " + (result ? "successfull" : "with problems") + ". vault ratio after clean-up " + cleanUpVaultCheck.collateralRatio)
+        return {
+            statusCode: result ? 200 : 500
+        }
+    }
+
+    await store.updateToState(VaultMaxiState.Start)
```

Comment:
why do we need the "start" state at all? Start and finished is equal right?
for me sounds more like we want states "Waiting", "doing transactions", "waiting for last transaction", "had error". cause thats the cases where we need to behave differently.

Reply:
yeah thought the same after setting it up, but thought it will be a good discussion starting point, as we most likely had completely different approach to implement this functionality. Will change to better states